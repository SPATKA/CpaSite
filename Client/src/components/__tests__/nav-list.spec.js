import React from "react"
import renderer from "react-test-renderer"
import NavList from "../nav-list"
import { render } from "@testing-library/react"
const config = require('../../../site-config');
config.taxServicesToggle = true;

describe("NavList", () => {
  const menuLinks = [
    {
      name: "Services",
      url: "/services",
      subLinks: [
        {
          name: "Tax Services",
          url: "/tax-services",
          isHidden: !config.taxServicesToggle,
          subLinks: [
            {
              name: "Tax Relief",
              url: "/tax-relief",
              isHidden: false,
              subLinks: [
                {
                  name: "IRS Audit Representation",
                  url: "/irs-audit-representation",
                  isHidden: false,
                  subLinks: [],
                },
              ]
            }
          ]
        }
      ],
    },
  ]


  it("renders correctly", () => {
    const tree = renderer
      .create(<NavList menuLinks={menuLinks}></NavList>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test.skip("should render given navigation list" , () => {
    const { getByText } = render(<NavList menuLinks={menuLinks}></NavList>)
    const servicesLink = getByText("Services")
    expect(servicesLink).toBeInTheDocument()
  })
})
