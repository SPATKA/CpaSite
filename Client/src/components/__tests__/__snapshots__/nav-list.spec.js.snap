// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList renders correctly 1`] = `
<ul
  className="linkContainer"
>
  <li
    className="linkItem"
  >
    <a
      className="navLinkText"
      href="/services"
    >
      Services
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-sm toggle-btn ps-2"
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
    <ul
      className="subLinkContainer1"
    >
      <li>
        <a
          className="navLinkText"
          href="/services/tax-services"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-left fa-sm toggle-btn pe-2"
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Tax Services
        </a>
        <ul
          className="subLinkContainer2"
        >
          <li>
            <a
              className="navLinkText"
              href="/services/tax-services/tax-relief"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-angle-left fa-sm toggle-btn pe-2"
                data-icon="angle-left"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
              Tax Relief
            </a>
            <ul
              className="subLinkContainer3"
            >
              <li>
                <a
                  className="navLinkText"
                  href="/services/tax-services/tax-relief/irs-audit-representation"
                >
                  IRS Audit Representation
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;
